app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCapturePresenterImpl.java

```diff
@@ -221,19 +221,21 @@ public void init() {
                                                     List<String> optionSets = new ArrayList<>();
                                                     for (FieldViewModel fieldViewModel : fields) {
                                                         String fieldSection = getFieldSection(fieldViewModel);
-                                                        if (!fieldMap.containsKey(fieldSection)) {
-                                                            fieldMap.put(fieldSection, new ArrayList<>());
-                                                        }
-                                                        fieldMap.get(fieldSection).add(fieldViewModel);
+                                                        if(!fieldSection.isEmpty() || sectionList.size() == 1) {
+                                                            if (!fieldMap.containsKey(fieldSection)) {
+                                                                fieldMap.put(fieldSection, new ArrayList<>());
+                                                            }
+                                                            fieldMap.get(fieldSection).add(fieldViewModel);
 
-                                                        if (fieldViewModel.optionSet() == null || !(fieldViewModel instanceof ImageViewModel)) {
-                                                            totalFields++;
-                                                        } else if (!optionSets.contains(fieldViewModel.optionSet())) {
-                                                            optionSets.add(fieldViewModel.optionSet());
-                                                            totalFields++;
+                                                            if (fieldViewModel.optionSet() == null || !(fieldViewModel instanceof ImageViewModel)) {
+                                                                totalFields++;
+                                                            } else if (!optionSets.contains(fieldViewModel.optionSet())) {
+                                                                optionSets.add(fieldViewModel.optionSet());
+                                                                totalFields++;
+                                                            }
+                                                            if (fieldViewModel instanceof UnsupportedViewModel || fieldViewModel instanceof DisplayViewModel)
+                                                                unsupportedFields++;
```

Comment:
it's okay, for next time @mmmateos  @ferdyrod  @Balcan  I'll push for having at least some unit testing regarding init method. It does not need to perform everything but  sectionList can easily be mocked and fieldMap asserted

Reply:
Remember, for future. If we don't have unit test. We'll have to add espresso which are expensive tests